clients/cfrestclient/rest_cloud_foundry_client_extended.go

```diff
@@ -1,65 +1,154 @@
+const defaultDomainsPathPattern = "/domains"
 
 type CloudFoundryRestClient struct {
 	baseclient.BaseClient
 	httpCloudFoundryClient *CloudFoundryClientExtended
+	cliConn                plugin.CliConnection
 }
 
-func NewCloudFoundryRestClient(host string, rt http.RoundTripper, tokenFactory baseclient.TokenFactory) CloudFoundryOperationsExtended {
-	t := baseclient.NewHTTPTransport(host, cfBaseUrl, rt)
+func NewCloudFoundryRestClient(cliConn plugin.CliConnection, rt http.RoundTripper, tokenFactory baseclient.TokenFactory) CloudFoundryOperationsExtended {
+	t := baseclient.NewHTTPTransport(getApiEndpoint(cliConn), cfBaseUrl, rt)
 	httpCloudFoundryClient := New(t, strfmt.Default)
-	return &CloudFoundryRestClient{baseclient.BaseClient{TokenFactory: tokenFactory}, httpCloudFoundryClient}
+	return &CloudFoundryRestClient{baseclient.BaseClient{TokenFactory: tokenFactory}, httpCloudFoundryClient, cliConn}
 }
 
-func (c CloudFoundryRestClient) GetSharedDomains() ([]models.SharedDomain, error) {
-	result := []models.SharedDomain{}
+func getApiEndpoint(cliConnection plugin.CliConnection) string {
+	api, err := cliConnection.ApiEndpoint()
+	if err != nil {
+		return ""
+	}
+	if strings.HasPrefix(api, "https://") {
+		api = strings.Replace(api, "https://", "", -1)
+	}
+	return api
+}
+
+func (c CloudFoundryRestClient) GetSharedDomains() ([]models.Domain, error) {
+	var result []models.Domain
 	var response *models.CloudFoundryResponse
 	var err error
-	var cloudFoundryUrlElememnts CloudFoundryUrlElements
-	pathPattern := defaultSharedDomainsPathPattern
+	var cloudFoundryUrlElements CloudFoundryUrlElements
+	pathPattern := defaultDomainsPathPattern
 	for pathPattern != "" {
-		response, err = c.getSharedDomainsInternal(cloudFoundryUrlElememnts)
+		response, err = c.getSharedDomainsInternal(cloudFoundryUrlElements)
 		if err != nil {
-			return []models.SharedDomain{}, err
+			return []models.Domain{}, models.HttpResponseError{Underlying: err}
 		}
 		result = append(result, toSharedDomains(response)...)
-		cloudFoundryUrlElememnts, err = getPathQueryElements(response)
+		cloudFoundryUrlElements, err = getPathQueryElements(response)
 		if err != nil {
-			return []models.SharedDomain{}, baseclient.NewClientError(err)
+			return []models.Domain{}, models.HttpResponseError{Underlying: baseclient.NewClientError(err)}
 		}
-		pathPattern = response.NextURL
+		pathPattern = response.Pagination.Next.Href
 	}
 
-	return result, err
+	return result, nil
+}
+
+func (c CloudFoundryRestClient) GetApplications(mtaId, currentSpaceGuid string) ([]models.CloudFoundryApplication, error) {
+	token, err := c.cliConn.AccessToken()
+	if err != nil {
+		return nil, fmt.Errorf("failed to retrieve access token: %s", err)
+	}
+	apiEndpoint, _ := c.cliConn.ApiEndpoint()
+	mtaIdHash := md5.Sum([]byte(mtaId))
+	mtaIdHashStr := hex.EncodeToString(mtaIdHash[:])
+
+	getAppsUrl := apiEndpoint + "/v3/apps?label_selector=mta_id=" + mtaIdHashStr + "&space_guids=" + currentSpaceGuid
```

Comment:
use fmt.Sprintf or something similar instead of string concatenation, apply the same for other methods as well